rdt/hyper_transformer.py

```diff
@@ -792,6 +794,52 @@ def fit_transform(self, data):
         self.fit(data)
         return self.transform(data)
 
+    def create_anonymized_columns(self, num_rows, column_names):
+        """Create the anonymized columns for this ``HyperTransformer``.
+
+        Generate the anonymized and text based columns that use regex expressions.
+
+        Args:
+            num_rows (int):
+                Number of rows to be created. Must be an integer greater than 0.
+            column_names (list):
+                List of column names to be created.
+
+        Returns:
+            pandas.DataFrame:
+                A data frame with the newly generated columns of the size ``num_rows``.
+        """
+        if not self._fitted or self._modified_config:
+            raise NotFittedError(self._NOT_FIT_MESSAGE)
```

Comment:
minor: let's just pull this chunk out into a `_validate_fitted` method that raises the error. This is used in 3 places now